app/V1Module/presenters/AssignmentSolutionsPresenter.php

```diff
@@ -520,6 +521,40 @@ public function actionDownloadSolutionArchive(string $id)
         $this->sendResponse(new StorageFileResponse($zipFile, "solution-{$id}.zip", "application/zip"));
     }
 
+    public function checkFiles(string $id)
+    {
+        $solution = $this->assignmentSolutions->findOrThrow($id);
+        if (!$this->assignmentSolutionAcl->canViewDetail($solution)) {
+            throw new ForbiddenRequestException("You cannot access the solution files metadata");
+        }
+    }
+
+    /**
+     * Get the list of submitted files of the solution.
+     * @GET
+     * @param string $id of assignment solution
+     * @throws ForbiddenRequestException
+     * @throws NotFoundException
+     */
+    public function actionFiles(string $id)
+    {
+        $solution = $this->assignmentSolutions->findOrThrow($id)->getSolution();
+        $entryPoint = $solution->getSolutionParams()->getVariable(ExecutionBox::$ENTRY_POINT_KEY);
```

Comment:
This feels weird, touching like really internal parts from frontend layer of the application...

Reply:
I thought about creating solution files view, but it seemed like an overkill at that moment. However, it is true that the code has perhaps overgrown the threshold and it would make sense now.